core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -254,16 +254,22 @@ import scala.util.control.NonFatal
   override val receive: Receive = regularReceive
 
   def regularReceive: Receive = LoggingReceive {
-    case Commit(tp, offset) =>
+    case Commit(tp, offset, flush) =>
       // prepending, as later received offsets most likely are higher
       commitMaps = tp -> offset :: commitMaps
       commitSenders = commitSenders :+ sender()
+      if (flush) {
+        requestDelayedPoll()
```

Comment:
As this is used for emergencies, we should consider calling `receivePoll` or `poll` immediately. That would improve chances the commit gets across to the broker before the consumer is shut down by stream error propagation.
`emergency` might be more accurate than `flush`.

Reply:
ok, this makes sense, I will change that